test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/ParsingSimpleTypeModelBinderTest.cs

```diff
@@ -0,0 +1,338 @@
+            Assert.Equal(32_000M, bindingContext.Result.Model);
+
+            var entry = Assert.Single(bindingContext.ModelState);
+            Assert.Equal("theModelName", entry.Key);
+            Assert.Empty(entry.Value.Errors);
+            Assert.Equal("32,000", entry.Value.AttemptedValue);
+            Assert.Equal(ModelValidationState.Unvalidated, entry.Value.ValidationState);
+        }
+
+        [Theory]
+        [InlineData(typeof(double))]
+        [InlineData(typeof(double?))]
+        public async Task BindModel_ThousandsSeparators_AreBoundToDoubles(Type destinationType)
+        {
+            // Arrange
+            var bindingContext = GetBindingContext(destinationType);
+            bindingContext.ValueProvider = new SimpleValueProvider(new CultureInfo("en-GB"))
+            {
+                { "theModelName", "32,000" }
+            };
+
+            var binder = new ParsingSimpleTypeModelBinder();
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(bindingContext.Result.IsModelSet);
+            Assert.Equal(32_000.0, bindingContext.Result.Model);
+
+            var entry = Assert.Single(bindingContext.ModelState);
+            Assert.Equal("theModelName", entry.Key);
+            Assert.Empty(entry.Value.Errors);
+            Assert.Equal("32,000", entry.Value.AttemptedValue);
+            Assert.Equal(ModelValidationState.Unvalidated, entry.Value.ValidationState);
+        }
+
+        [Theory]
+        [InlineData(typeof(float))]
+        [InlineData(typeof(float?))]
+        public async Task BindModel_ThousandsSeparators_AreBoundToFloats(Type destinationType)
+        {
+            // Arrange
+            var bindingContext = GetBindingContext(destinationType);
+            bindingContext.ValueProvider = new SimpleValueProvider(new CultureInfo("en-GB"))
+            {
+                { "theModelName", "32,000" }
+            };
+
+            var binder = new ParsingSimpleTypeModelBinder();
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(bindingContext.Result.IsModelSet);
+            Assert.Equal(32_000.0F, bindingContext.Result.Model);
+
+            var entry = Assert.Single(bindingContext.ModelState);
+            Assert.Equal("theModelName", entry.Key);
+            Assert.Empty(entry.Value.Errors);
+            Assert.Equal("32,000", entry.Value.AttemptedValue);
+            Assert.Equal(ModelValidationState.Unvalidated, entry.Value.ValidationState);
+        }
+
+        [Fact]
+        public async Task BindModel_ValidValueProviderResultWithProvidedCulture_ReturnsModel()
+        {
+            // Arrange
+            var bindingContext = GetBindingContext(typeof(double));
+            bindingContext.ValueProvider = new SimpleValueProvider(new CultureInfo("fr-FR"))
+            {
+                { "theModelName", "12,5" }
```

Comment:
At least `fr-FR` does not have a thousands separator. I'll look at a few other cultures to find something that does both.